src/Integration.Vsix.UnitTests/Analysis/IssueHandlerTests.cs

```diff
@@ -0,0 +1,271 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.Text;
+using Moq;
+using SonarLint.VisualStudio.Core.Suppression;
+using SonarLint.VisualStudio.Integration.Vsix;
+using SonarLint.VisualStudio.Integration.Vsix.Analysis;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using static SonarLint.VisualStudio.Integration.Vsix.Analysis.IssueConsumerFactory;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Analysis
+{
+    [TestClass]
+    public class IssueHandlerTests
+    {
+        [TestMethod]
+        public void WhenIssuesAreFound_FilterIsApplied_NewSnapshotIsPublished()
```

Comment:
These tests were all moved from `TextBufferIssueTrackerTests` and the scaffolding modified appropriately.